### ISSUE TITLE
LWM2M Bootstrap towards Leshan Server fails.        ( leshan.eclipseprojects.io/ )

### ISSUE THREAD

**Issue by mpavlovic0815** [eclipse/leshan#819]
Hello,

I am facing frequent issues with bootstrapping from a remote LWM2M Server towards the Leshan server.
I can see handshake failures reported from the Leshan server, even if the direct connect with the same credentials works.
Have you faced any issues last 2 weeks ?

Thanks!
Cheers,
MP

**Comment by sbernard31** [comment]
What do you call "the Leshan Server" ?
Is it the Leshan Server Sandbox available at : https://leshan.eclipseprojects.io/#/clients ?

Which kind of handshake failures did you face ?

**Comment by sbernard31** [comment]
Now you change the issue title, I get that you are using the sandbox.

So if you want more information about the handshake failure, I need a more data. Like a tcpdump/wireshark capture.

**Comment by sbernard31** [comment]
By the way the sandbox is automatically restarted to update it (system updates or leshan updates) and so there is no guaranty it is always up. If you need a stable environment you should consider to host it yourself.

**Comment by mpavlovic0815** [comment]
Hi,
thanks for getting back. It is the  : https://leshan.eclipseprojects.io/#/clients  [23.97.187.154]

**Comment by sbernard31** [comment]
Does your device use X509 to authenticate itself ?

**Comment by mpavlovic0815** [comment]
No, just the configured Key.

**Comment by sbernard31** [comment]
You mean you are using PreSharedKey ?

**Comment by sbernard31** [comment]
I looked at all the sandbox logs of this month and so only handshake issue I see is about certificate
(I probably found a bug which prevent to connect to leshan sandbox using x509) but nothing about PSK.

So If you can provide a wireshark/tcpdump capture :)

**Comment by mpavlovic0815** [comment]
[8988228066700007078.zip](https://github.com/eclipse/leshan/files/4344787/8988228066700007078.zip)

**Comment by sbernard31** [comment]
Ok, I just looked at it and your device uses only deprecated CBC block ciphers.

As this ciphers are not consider as safe enough we deactivate it by default. 
I would not like to activate it on the sandbox as I would not encourage to use this ciphers.

But If you run the sandbox by yourself you can activate old cipher option with : 
```
-oc                            activate support of old/deprecated cipher suites.
```

**Comment by sbernard31** [comment]
(I fixed the x509 issue : commit 25413f2)

Could we close this issue ?

**Comment by mpavlovic0815** [comment]
Thanks for Support.
